kayenta-core/src/main/java/com/netflix/kayenta/metrics/SynchronousQueryProcessor.java

```diff
@@ -81,10 +87,20 @@ public String executeQuery(String metricsAccountName,
         registry.counter(queryId.withTag("retries", retries + "")).increment();
         metricSetList = metricsService.queryMetrics(metricsAccountName, canaryConfig, canaryMetricConfig, canaryScope);
         success = true;
-      } catch (IOException | UncheckedIOException | RetrofitError | RetryableQueryException e) {
+      } catch (RetrofitError e) {
+
+        boolean retryable = isRetryable(e);
+        if (retryable) {
+          retries++;
+          if (retries >= retryConfiguration.getAttempts())
+            throw e;
+          log.warn("Retrying metric service query");
```

Comment:
This log message won't give the user much to go on when troubleshooting. Can you include some details that will help when tracing requests?

Reply:
fixed error message for both cases